### PR TITLE
Molt 124 refresh ligand library

### PR DESCRIPTION
**Pull request by mag8224**
### Description
Added API call to the Protein Data bank to get all the PDB Ids of the currently aligned protein's EC Class.

### Checklist
- [x] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/main/java/org/moltimate/moltimatebackend/service/LigandService.java [hunk 1]
```diff
@@ -112,4 +134,67 @@ public String getEcNumber(Structure structure) throws InvalidPdbIdException {
...
+        log.info("GET Response Code :: " + responseCode);
+        StringBuffer response = new StringBuffer();
+        if (responseCode == HttpURLConnection.HTTP_OK) { // success
+            BufferedReader in = new BufferedReader(new InputStreamReader(httpURLConnection.getInputStream()));
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                response.append(inputLine);
+            } in.close();
+        } else {
+            log.error("GET request not worked");
```
Comment (arp6695) [inline src/main/java/org/moltimate/moltimatebackend/service/LigandService.java:174]: Typo. "GET Request did not work."

File: src/main/java/org/moltimate/moltimatebackend/service/LigandService.java [hunk 2]
```diff
@@ -112,4 +134,67 @@ public String getEcNumber(Structure structure) throws InvalidPdbIdException {
...
+     * call the Protein Data bank's Search api to get the PDB Ids from the input EC Class
+     * @param ecNumber
+     * @return List of pdbIds in the EC class
+     */
+    public List<String> getPdbIdsFromEcClass(String ecNumber) throws IOException {
+        List<String> pdbIds = new ArrayList<>();
+
+        String USER_AGENT = "Mozilla/5.0";
+
+        String baseURL = "https://search.rcsb.org/rcsbsearch/v1/query?json=";
```
Comment (arp6695) [inline src/main/java/org/moltimate/moltimatebackend/service/LigandService.java:147]: I've only partially read Herbert's post; are we certain that this is not the legacy API?

**Review by arp6695 — APPROVED** [review]
Some typos, but otherwise verified.

**Review by mev1482 — APPROVED** [review]

File: src/main/java/org/moltimate/moltimatebackend/service/LigandService.java (hunk 2 above)
Comment (mag8224) [reply src/main/java/org/moltimate/moltimatebackend/service/LigandService.java:147]: Yes we are sure this isn't the legacy API.

**Pull request merged**